### PR TITLE
[WIP] Adicionando notebooks do dataset

### PR DESCRIPTION
**Pull request by andrewlucasgs**
- [x] Adiconar notebook para scraping tweets por pesquisa
- [x] Adiconar notebook para scraping tweets por usuário
- [ ] Adicionar notebook para optimização do dataset

### CODE REVIEW

**Comment by andrewlucasgs** [comment]
Estou trabalhando no pré-processamento dos tweets coletados, quando eu terminar, acredito que poderemos mergar na master, e começar a trabalhar na rede neural.

Eu encontrei alguns artigos sobre isso:
https://aclweb.org/anthology/W18-0609

**Comment by zarbielli** [comment]
Adicionei tweets de 241 usuário, até agora temos 843212 mil tweets de dataset. Fechamos a issue #7

**Comment by andrewlucasgs** [comment]
Criei o modelo, mas o kernel do jupyter morre toda vez que tento rodar, estou buscando uma solução

**Comment by andrewlucasgs** [comment]
Reduzi o número de tweets no dataset, e rodou, mas está com accuracia 0, acredito que temos que mudar a abordagem dos dados

Layer (type)                 Output Shape              Param #   
=================================================================
embedding_1 (Embedding)      (None, 100, 100)          14172300  
_________________________________________________________________
conv1d_1 (Conv1D)            (None, 96, 128)           64128     
_________________________________________________________________
max_pooling1d_1 (MaxPooling1 (None, 48, 128)           0         
_________________________________________________________________
flatten_1 (Flatten)          (None, 6144)              0         
_________________________________________________________________
dense_1 (Dense)              (None, 128)               786560    
_________________________________________________________________
dense_2 (Dense)              (None, 1)                 129       
=================================================================
Total params: 15,023,117
Trainable params: 850,817
Non-trainable params: 14,172,300
_________________________________________________________________
None
WARNING:tensorflow:From /usr/local/anaconda/lib/python3.7/site-packages/tensorflow/python/ops/math_ops.py:3066: to_int32 (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
Train on 167523 samples, validate on 41880 samples
Epoch 1/20
 - 165s - loss: nan - acc: 0.0154 - val_loss: nan - val_acc: 0.0000e+00
Epoch 2/20
 - 164s - loss: nan - acc: 0.0000e+00 - val_loss: nan - val_acc: 0.0000e+00
Epoch 3/20
 - 157s - loss: nan - acc: 0.0000e+00 - val_loss: nan - val_acc: 0.0000e+00
Epoch 4/20
 - 149s - loss: nan - acc: 0.0000e+00 - val_loss: nan - val_acc: 0.0000e+00

**Comment by gdeusdara** [comment]
Fiz com o tamanho original do dataset em um computador melhor, e o kernel conseguiu aguentar, mas ficou com a acuracia 0 do mesmo jeito

Layer (type)                 Output Shape              Param #   
=================================================================
embedding_1 (Embedding)      (None, 100, 100)          27871000  
_________________________________________________________________
conv1d_1 (Conv1D)            (None, 96, 128)           64128     
_________________________________________________________________
max_pooling1d_1 (MaxPooling1 (None, 48, 128)           0         
_________________________________________________________________
flatten_1 (Flatten)          (None, 6144)              0         
_________________________________________________________________
dense_1 (Dense)              (None, 128)               786560    
_________________________________________________________________
dense_2 (Dense)              (None, 1)                 129       
=================================================================
Total params: 28,721,817
Trainable params: 850,817
Non-trainable params: 27,871,000
_________________________________________________________________
None
WARNING:tensorflow:From /usr/lib/python3.7/site-packages/tensorflow/python/ops/math_ops.py:3066: to_int32 (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
Train on 487523 samples, validate on 121880 samples
Epoch 1/20
 - 414s - loss: nan - acc: 4.9434e-04 - val_loss: nan - val_acc: 0.0000e+00
Epoch 2/20
 - 352s - loss: nan - acc: 0.0000e+00 - val_loss: nan - val_acc: 0.0000e+00
Epoch 3/20
 - 338s - loss: nan - acc: 0.0000e+00 - val_loss: nan - val_acc: 0.0000e+00
Epoch 4/20

**Comment by andrewlucasgs** [comment]
Tentei usar uma camada de LSTM mas sem sucesso também. Vou tentar reduzir o escopo e tratar os tweets.

Please use `rate` instead of `keep_prob`. Rate should be set to `rate = 1 - keep_prob`.
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
embedding_3 (Embedding)      (None, 100, 100)          14172300  
_________________________________________________________________
conv1d_3 (Conv1D)            (None, 100, 32)           9632      
_________________________________________________________________
max_pooling1d_3 (MaxPooling1 (None, 50, 32)            0         
_________________________________________________________________
dropout_1 (Dropout)          (None, 50, 32)            0         
_________________________________________________________________
lstm_1 (LSTM)                (None, 300)               399600    
_________________________________________________________________
dropout_2 (Dropout)          (None, 300)               0         
_________________________________________________________________
dense_3 (Dense)              (None, 1)                 301       
=================================================================
Total params: 14,581,833
Trainable params: 409,533
Non-trainable params: 14,172,300
_________________________________________________________________
None
Train on 167523 samples, validate on 41880 samples
Epoch 1/20
 - 585s - loss: nan - acc: 0.0344 - val_loss: nan - val_acc: 0.0000e+00
Epoch 2/20
 - 603s - loss: nan - acc: 0.0000e+00 - val_loss: nan - val_acc: 0.0000e+00
Epoch 3/20
 - 649s - loss: nan - acc: 0.0000e+00 - val_loss: nan - val_acc: 0.0000e+00
Epoch 4/20

**Comment by andrewlucasgs** [comment]
Criei um dataset menor, e mais confiavel, e usando a mesma rede neural da ultima tentativa, consegui acurácia  de 67.266592, vou testar algumas modificações, acho que ainda da para melhorar

**Comment by gdeusdara** [comment]
> Criei um dataset menor, e mais confiavel, e usando a mesma rede neural da ultima tentativa, consegui acurácia de 67.266592, vou testar algumas modificações, acho que ainda da para melhorar

Muito boa!

**Comment by mansogf** [comment]
Foi feito as devidas correções do dataset e nas camadas.

Foi utilizado como referência o artigo que o @andrewlucasgs comentou lá em cima (https://aclweb.org/anthology/W18-0609). 

A acurácia que estava ficando quase que permanentemente em 65% agora alcançou 80-88%.

**Comment by andrewlucasgs** [comment]
Essa branch foi reformulada e movida para dev_new branch, então esse PR será recusado, mas a branch será mantida como aprendizado.
